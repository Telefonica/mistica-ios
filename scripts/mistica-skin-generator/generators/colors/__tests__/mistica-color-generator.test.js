import fs from 'fs';
import {resolve} from 'path';
import {reduceColors} from '../reduce-colors';
import {generateMisticaColor} from '../mistica-color-generator';

test('generateMisticaColor', () => {
    const getTokensFromFile = (file) => JSON.parse(fs.readFileSync(resolve(__dirname, file), 'utf-8'));
    const brandsWithTokens = [
        {
            brand: {
                id: 'brand1',
                prefix: 'Brand1',
                misticaDesignFileName: 'brand-1',
            },
            tokens: getTokensFromFile('brand1-tokens.json'),
        },
        {
            brand: {
                id: 'brand2',
                prefix: 'Brand2',
                misticaDesignFileName: 'brand-2',
            },
            tokens: getTokensFromFile('brand2-tokens.json'),
        },
    ];

    const colors = reduceColors(brandsWithTokens);
    const result = generateMisticaColor(colors);

    expect(result).toMatchSnapshot();
});
