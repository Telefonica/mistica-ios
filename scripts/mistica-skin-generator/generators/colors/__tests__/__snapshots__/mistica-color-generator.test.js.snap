// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateMisticaColor 1`] = `
"
// Generated using Make
// DO NOT EDIT

import UIKit

public enum MisticaColor {
    case solid(UIColor)
    case gradient(MisticaGradient)
}

public extension MisticaColor {
    static var backgroundBrand: MisticaColor {
        MisticaConfig.currentColors.backgroundBrand
    }
    static var backgroundContainerBrand: MisticaColor {
        MisticaConfig.currentColors.backgroundContainerBrand
    }
    static var cardContentOverlay: MisticaColor {
        MisticaConfig.currentColors.cardContentOverlay
    }
}
"
`;
